bin/celeste.jl

```diff
@@ -27,7 +33,175 @@ field, using only that field.
 The `score-field` subcommand is not yet implemented for the new API.
 """
 
+"""
+An area of the sky subtended by `ramin`, `ramax`, `decmin`, and `decmax`.
+"""
+type SkyArea
+    ramin::Float64
+    ramax::Float64
+    decmin::Float64
+    decmax::Float64
+    nra::Int64
+    ndec::Int64
+end
+
+"""
+Given a SkyArea that is to be divided into `skya.nra` x `skya.ndec` patches,
+return the `i`th patch. `i` is a linear index between 1 and
+`skya.nra * skya.ndec`.
+
+This function assumes a cartesian (rather than spherical) coordinate system!
+"""
+function sky_subarea(skya::SkyArea, i)
+    global wira, widec
+    ix, iy = ind2sub((skya.nra, skya.ndec), i)
+
+    return (skya.ramin + (ix - 1) * wira,
+            min(skya.ramin + ix * wira, skya.ramax),
+            skya.decmin + (iy - 1) * widec,
+            min(skya.decmin + iy * widec, skya.decmax))
+end
+
+
+"""
+Divide the area of sky subtended by `ramin`, `ramax`, `decmin`, and `decmax`
+into patches of `wira`X`widec`. Each such patch is a work item. Use Dtree to
+distribute these to nodes. Within the node, run inference on each patch
+received from Dtree.
+"""
+function celeste_skyarea(ramin, ramax, decmin, decmax, outdir)
+    # ---
+    if dt_nodeid == 1
+        println("celeste -- running on $dt_nnodes nodes (system clock speed is $cpu_hz Hz)")
+	println("  sky area: $(ramax - ramin) X $(decmax - decmin)")
+    end
+
+    n_infers = 0
+    n_fits = 0
+    tm_infer = tm_fldids = tm_rdprim = tm_filcat = tm_rdimg = tm_imp = tm_psf = tm_fit = tm_out = tm_done = 0.0
```

Comment:
There are a lot of line overruns and tab misalignments -- I won't comment each one.